added support to get list of users

moriatuch opened the pull request.

Commit:
added support to get list of users

martinohmann commented on api/types.go:
```diff
@@ -155,6 +155,13 @@ type UserGroup struct {
 	AttributeGroupID string   `json:"attribute_group_id,omitempty"`
 }
 
+type Users struct {
```
Since this represents a single user this should use the singular form `User`. Also this should have a doc comment for the API documentation. Suggestion:

```
// User who can login to Site24x7 and receive instant notifications about outages.
```

martinohmann commented on client.go:
```diff
@@ -147,6 +148,9 @@ func (c *client) ThresholdProfiles() endpoints.ThresholdProfiles {
 func (c *client) UserGroups() endpoints.UserGroups {
 	return endpoints.NewUserGroups(c.restClient)
 }
+func (c *client) Users() endpoints.Users {
```
Please add an empty line before this method and add a doc comment to be consistent with the other methods. Suggestion:

```
// Users implements Client.
```

martinohmann commented on api/endpoints/testdata/fixtures/responses/list_users.json:
```diff
@@ -0,0 +1,31 @@
+{
```
Small nitpick, feel free to ignore: the formatting is a little bit inconsistent throughout the file, maybe use a JSON formatter to autoformat it.

martinohmann reviewed (changes requested):
Hi @moriatuch,

thanks for the addition! I left some minor comments.

The tests are failing right now because you missed to also update the fake client implementation in `fake/client.go` with the added endpoints and also create `api/endpoints/fake/users.go` with a fake implementation of the users endpoints.

After this is addressed we'd be happy to merge it.

Commit:
update Pr according to  martinohmann's comments
- formatting the json of users list
- change the type from users to user
- added some commemts

martinohmann commented:
@moriatuch did you already have time to look into adding the fake client implementation as noted above? Without this the PR cannot be merged due to the failing build.

martinohmann commented:
Closing due to inactivity.
